File: js/task1/S1_IF_03.js
// S1_IF_03

function normalizeName(input) {
    // check if falsy (empty, null, etc)
    if (!input) {
        return "Anonymous";
    }
    return input.trim();
}

console.log(normalizeName(""));
console.log(normalizeName(null));
console.log(normalizeName("  John  "));

File: python/task1/S1_IF_03.py
# S1_IF_03

def normalize_name(x):
    if not x: # checks for None, empty string, etc.
        return "Anonymous"
    
    cleaned = x.strip()
    if cleaned == "":
        return "Anonymous"
        
    return cleaned

print(normalize_name(None))
print(normalize_name("  John  "))
print(normalize_name("   "))